ojsErcPlugin.inc.php

```diff
@@ -0,0 +1,439 @@
+
+				$contextId = Application::get()->getRequest()->getContext()->getId();
+				$baseUrlErcFromSettings = $this->getSetting($contextId, 'serverURL'); 
+
+				$baseUrlErcStandard= "https://o2r.uni-muenster.de/api/v1/"; 
+
+				// if there is no url available from the plugin settings the standard url is used 
+				if ($baseUrlErcFromSettings === null || $baseUrlErcFromSettings === '') {
+					$baseUrlErc = $baseUrlErcStandard;
+				}
+				else {
+					$baseUrlErc = $baseUrlErcFromSettings;
+				}
+
+				preg_match('/"baseUrl":\s"[^,]*"/', $readConfigFile, $configOld);        
+				$configNew = '"baseUrl": "' . $baseUrlErc . '"'; 
+				$adaptedConfig = str_replace($configOld[0], $configNew, $readConfigFile);
+
+				$adaptedOjsView = str_replace("false", "true", $adaptedConfig);
+
+				file_put_contents($pathConfigJs, $adaptedOjsView);
+				fclose($rawConfigFile);
+
+				/*
+				Update the index.html of the build in terms of the users path for the css-, js-, logo-, and manifest-file(s). 
+				*/
+				$request = Application::get()->getRequest();
+				$baseUrl = $request->getBaseUrl();
+
+				$pathHtml = $this->getPluginPath() . '/build/index.html'; 
+
+				$rawHtmlFile = fopen($pathHtml, "r+");
+				$readHtmlFile = fread($rawHtmlFile, filesize($pathHtml)); 
+
+				preg_match('"\.\/static\/css\/[^=]*\.chunk\.css"', $readHtmlFile, $staticCssOld);
+				$staticCssNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($staticCssOld[0], 1); 
+				$adaptedCss = str_replace($staticCssOld[0], $staticCssNew, $readHtmlFile);
+					
+				preg_match('"\.\/static\/css\/main[^=]*\.chunk\.css"', $readHtmlFile, $staticCssMainOld);
+				$staticCssMainNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($staticCssMainOld[0], 1); 
+				$adaptedCssMain = str_replace($staticCssMainOld[0], $staticCssMainNew, $adaptedCss);
+
+				preg_match('"\.\/static\/js\/[^=]*\.chunk\.js"', $readHtmlFile, $staticJsOld);
+				$staticJsNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($staticJsOld[0], 1); 
+				$adaptedJs = str_replace($staticJsOld[0], $staticJsNew, $adaptedCssMain);
+
+				preg_match('"\.\/static\/js\/main[^=]*\.chunk\.js"', $readHtmlFile, $staticJsMainOld);
+				$staticJsMainNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($staticJsMainOld[0], 1); 
+				$adaptedMainJs = str_replace($staticJsMainOld[0], $staticJsMainNew, $adaptedJs);
+
+				preg_match('"\.\/logo\.png"', $readHtmlFile, $logoOld);
+				$logoNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($logoOld[0], 1); 
+				$adaptedLogo = str_replace($logoOld[0], $logoNew, $adaptedMainJs);
+
+				preg_match('"\.\/manifest\.json"', $readHtmlFile, $manifestOld);
+				$manifestNew = $baseUrl . '/' . $this->getPluginPath() . '/build' . substr($manifestOld[0], 1); 
+				$adaptedManifest = str_replace($manifestOld[0], $manifestNew, $adaptedLogo);
+
+				preg_match('"\.\/config\.js"', $readHtmlFile, $configOld);
```

Comment:
Can you encapsulate this in a function? Looks very repetitive.

Reply:
Yeah is done here: https://github.com/o2r-project/ojs-erc-plugin/pull/1/commits/a23c3354f9655c3bede6e6a2bf77352e7c441df2